MetaMask/metamask-extension
# Create new inline contract-details field for easier verification

## **Description**

Verifying signatures sources is quite important and moving the verification to a modal removes the context of what the verifying address is doing. Moving it inline and potentially linking in information from the associated NFT or ERC-20 contracts can better alert users to what they are doing and allow for super users to have better context when parsing signature transactions. 

A signature associated with a low market cap coin while there is another with a high market cap coin can be confusing and a huge difference.

## **Related issues**

Fixes:

## **Manual testing steps**

1. Attempt to sign a typed message
2. See the verifying address inline rather than in a modal
3. Allows for 1-click verification and (in the future) better display if you are signing against a ERC 20 / 721 token.

## **Screenshots/Recordings**

### **Before**


<img width="366" alt="Screenshot 2024-02-14 at 5 00 22 PM" src="https://github.com/MetaMask/metamask-extension/assets/835896/4532137c-5907-44c4-87d8-e33fccda2c5d">

### **After**

<img width="372" alt="Screenshot 2024-02-14 at 4 28 28 PM" src="https://github.com/MetaMask/metamask-extension/assets/835896/d32afc6a-32af-4ddf-bb46-13f5b998cf4a">


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
